File: backend/app/schemas/jobs.py
from datetime import date

from pydantic import BaseModel


class JobSearchRequest(BaseModel):
    query: str


class JobCard(BaseModel):
    id: str
    role: str
    company: str
    location: str
    salary_range: str | None = None
    deadline: date | None = None
    url: str
    source_platform: str | None = None
    fit_score: int
    fit_reasoning: str
    missing_skills: list[str] = []


class JobSearchResponse(BaseModel):
    results: list[JobCard]
    source: str
    total: int


class FitScoreRequest(BaseModel):
    jd_text: str


class ScoreBreakdown(BaseModel):
    skill_match: int
    semantic_match: int
    experience_match: int


class FitScoreResponse(BaseModel):
    score: int
    breakdown: ScoreBreakdown
    explanation: str


class CoverLetterRequest(BaseModel):
    role: str
    company: str
    jd_summary: str  # fit_reasoning or a short description of the role


class CoverLetterResponse(BaseModel):
    cover_letter: str

File: frontend/store/jobs.ts
import { create } from "zustand";

export interface JobCard {
  id: string;
  role: string;
  company: string;
  location: string;
  salary_range: string | null;
  deadline: string | null;
  url: string;
  source_platform: string | null;
  fit_score: number;
  fit_reasoning: string;
  missing_skills: string[];
}

interface JobsState {
  results: JobCard[];
  source: string | null;
  total: number;
  isSearching: boolean;
  query: string;
  setResults: (results: JobCard[], source: string | null, total: number) => void;
  setSearching: (v: boolean) => void;
  setQuery: (q: string) => void;
  clear: () => void;
}

export const useJobsStore = create<JobsState>()((set) => ({
  results: [],
  source: null,
  total: 0,
  isSearching: false,
  query: "",
  setResults: (results, source, total) => set({ results, source, total }),
  setSearching: (v) => set({ isSearching: v }),
  setQuery: (q) => set({ query: q }),
  clear: () => set({ results: [], source: null, total: 0, query: "" }),
}));
